src/Avalonia.Diagnostics/Diagnostics/ViewModels/MainViewModel.cs

```diff
@@ -76,7 +76,20 @@ public bool ShouldVisualizeDirtyRects
             get => _shouldVisualizeDirtyRects;
             set
             {
-                ((TopLevel)_root).Renderer.DrawDirtyRects = value;
+                var changed = true;
+                if (_root is TopLevel topLevel && topLevel.Renderer is { })
+                {
+                    topLevel.Renderer.DrawDirtyRects = value;
+                }
+                else if (_root is Controls.Application app && app.RendererRoot is { })
+                {
+                    app.RendererRoot.DrawDirtyRects = value;
```

Comment:
Shouldn't it draw dirty rects for all windows except devtools one?

Reply:
I will merge this PR to fix the issue. But can you create a new one, if you agree?